shell/platform/darwin/ios/framework/Headers/FlutterEngine.h

```diff
@@ -119,6 +121,15 @@ FLUTTER_EXPORT
  */
 - (BOOL)runWithEntrypoint:(NSString*)entrypoint libraryURI:(NSString*)uri;
 
+/**
+ * Destroy running context for an engine.
+ *
+ * If you create a FlutterEngine with allowHeadlessExecution set to YES,
+ * call destroyContext explicitly when you want to release the engine.
+ *
+ */
+- (void)destroyContext;
```

Comment:
Do we need to expose this?  Is it enough to just call it from `dealloc`?

Is there something about this that prevents the Engine from going out of scope locally?

Reply:
When developer uses
`- (instancetype)initWithName:(NSString*)labelPrefix project:(FlutterDartProject*)projectOrNil`
allowHeadlessExecution is set to YES and the engine will not reset the `shell` & `threadhost` when the attatched viewcontroller is dealloced.
```
- (void)notifyViewControllerDeallocated {
  if (!_allowHeadlessExecution) {
    [self destroyContext];
  }
}
```

So we need to expose this api, so that developer can manually call `destroyContext` and release the engine instance when needed.

Reply:
Why can't we just call `destroyContext` from `FlutterEngine::dealloc`?